### DISCUSSION TITLE
Workspaces API

### DISCUSSION THREAD

**Discussion by TheBlueOompaLoompa** [zen-browser/desktop#1215] [Ideas]
Allow extensions to interact with workspaces, (or document it if it exists) so extensions can be created to integrate with it. For example, some people want tabs inside specific workspaces to default to a container tab. Being able to extend it might allow for other things such as workspace specific features, settings, theming, etc.

**Comment by LimitLost** [comment]
I would also love to lock specific tabs to certain workspaces, for example lock out youtube tabs from work workspace. When youtube tab will be open I want to either show a full page warning or change the current workspace
